src/reducers/layout.js

```diff
@@ -17,6 +22,8 @@ export default (state = INITIAL_STATE, { type, payload }) => {
       return state.setIn(['tools', 'open'], false);
     case SET_TOOLS_WIDTH:
       return state.setIn(['tools', 'width'], payload);
+    case CHANGE_SIDE_BAR_VIEW_STATE:
+      return state.setIn(['sideBarState'], payload);
```

Comment:
I agree with @pyphilia. In fact, I would follow the strategy followed by tools. With a depth of two levels.